src/scripttabs.cpp

```diff
@@ -108,7 +108,7 @@ TYPE_SYMBOL asTypeTable[] =
 	{ "",	(INTERP_TYPE)ST_POINTER_STRUCTSTAT, AT_SIMPLE, NULL, NULL },	//for NULLSTRUCTURESTAT
 
 	/* This final entry marks the end of the type list */
-	{ "END OF TYPE LIST", (INTERP_TYPE)NULL,		AT_SIMPLE,	NULL,	NULL },
+	{ "END OF TYPE LIST", (INTERP_TYPE)0,		AT_SIMPLE,	NULL,	NULL },
```

Comment:
Ah, I didn't notice that it wasn't of pointer type.

As for the code still being broken, with regards to your first point: casting/converting enums is not undefined behaviour it is very well-defined behaviour (given that both the C and C++ specs fully specify how their numerical value gets established).

As for your second comment, it's not the use of C arrays, but rather the reliance on sentinel values (zero) to detect end-of-data that's the problem. Using `begin() == array`, `end() == array + sizeof(array[0])` would work perfectly. In fact that would be better than any C++ container until a container that's fully initializeable by an `initializer_list` as `constexpr` comes into existence. The static array has the advantage of everything being able to live in `.rodata`. (Of course right now it doesn't due to const incorrectness).